src/streamlink/plugins/vlive.py

```diff
@@ -9,48 +9,60 @@
 
 
 class Vlive(Plugin):
-    _url_re = re.compile(r"https?://(?:www.)vlive\.tv/video/(\d+)")
-    _video_status = re.compile(r'oVideoStatus = (.+)<', re.DOTALL)
-    _video_init_url = "https://www.vlive.tv/video/init/view"
+    _url_re = re.compile(r"https?://(?:www.)vlive\.tv/(?P<format>video|post)/(?P<id>[0-9\-]+)")
+    _page_info = re.compile(r'window\.__PRELOADED_STATE__\s*=\s*({.*})\s*<', re.DOTALL)
+    _playinfo_url = "https://www.vlive.tv/globalv-web/vam-web/old/v3/live/%s/playInfo"
 
     @classmethod
     def can_handle_url(cls, url):
         return cls._url_re.match(url) is not None
 
-    @property
-    def video_id(self):
-        return self._url_re.match(self.url).group(1)
-
     def _get_streams(self):
-        vinit_req = self.session.http.get(self._video_init_url,
-                                          params=dict(videoSeq=self.video_id),
-                                          headers=dict(referer=self.url))
-        if vinit_req.status_code != 200:
-            raise PluginError('Could not get video init page (HTTP Status {})'
-                              .format(vinit_req.status_code))
+        video_page = self.session.http.get(self.url, headers=dict(referer=self.url))
+        if video_page.status_code != 200:
+            raise PluginError('Could not get video page (HTTP Status {})'
+                              .format(video_page.status_code))
+
+        page_info_js = self._page_info.search(video_page.text)
+        if not page_info_js:
+            raise PluginError('Could not find page info')
```

Comment:
Unavailable streams are not supposed to raise a PluginError. Just return None or an empty dict. PluginErrors are meant for actual errors. Same issue below...